src/network/netdev/wireguard.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+/***
+  This file is part of systemd.
+
+  Copyright 2016 Jörg Thalheim <joerg@thalheim.io>
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+typedef struct Wireguard Wireguard;
+
+#include "netdev.h"
+#include "sd-resolve.h"
+#include "wireguard-netlink.h"
+
+#ifndef IFNAMSIZ
+#define IFNAMSIZ 16
+#endif
+
+typedef struct WireguardIpmask {
+        uint16_t family;
+        union {
+                struct in_addr ip4;
+                struct in6_addr ip6;
+        };
+        uint8_t cidr;
+
+        LIST_FIELDS(struct WireguardIpmask, ipmasks);
+} WireguardIpmask;
+
+typedef struct WireguardPeer {
+        uint8_t public_key[WG_KEY_LEN];
+        uint8_t preshared_key[WG_KEY_LEN];
+        uint32_t flags;
+
+        union {
+          struct sockaddr addr;
+          struct sockaddr_in addr4;
+          struct sockaddr_in6 addr6;
+        } endpoint; /* Get/Set */
```

Comment:
use `union sockaddr_union`, even if in theory it could be larger than what we need here...

also, 8ch indenting